// sanity-plugin-mux-input/build/components/FileInputButton.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.FileInputButton = void 0;

var _ui = require("@sanity/ui");

var _lodash = require("lodash");

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _excluded = ["onSelect", "type"];

var _templateObject, _templateObject2;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var HiddenInput = _styledComponents.default.input(_templateObject || (_templateObject = _taggedTemplateLiteral(["\n  overflow: hidden;\n  width: 0.1px;\n  height: 0.1px;\n  opacity: 0;\n  position: absolute;\n  z-index: -1;\n"])));

var Label = _styledComponents.default.label(_templateObject2 || (_templateObject2 = _taggedTemplateLiteral(["\n  position: relative;\n"])));

var handleSelect = _ref => {
  var event = _ref.event,
      onSelect = _ref.onSelect;

  if (onSelect) {
    onSelect(event.target.files);
  }
};

var handleButtonClick = inputRef => {
  if (inputRef && inputRef.current) {
    inputRef.current.click();
  }
};

var FileInputButton = _ref2 => {
  var onSelect = _ref2.onSelect,
      _ref2$type = _ref2.type,
      type = _ref2$type === void 0 ? 'file' : _ref2$type,
      props = _objectWithoutProperties(_ref2, _excluded);

  var _inputId = _react.default.useRef((0, _lodash.uniqueId)('FileSelect'));

  var inputRef = _react.default.useRef(null);

  return /*#__PURE__*/_react.default.createElement(Label, {
    htmlFor: _inputId.current
  }, /*#__PURE__*/_react.default.createElement(HiddenInput, {
    ref: inputRef,
    tabindex: "0",
    type: type,
    id: _inputId.current,
    onChange: event => handleSelect({
      event,
      onSelect
    }),
    value: ""
  }), /*#__PURE__*/_react.default.createElement(_ui.Button, _extends({
    onClick: () => handleButtonClick(inputRef),
    mode: "default",
    tone: "primary",
    style: {
      width: '100%'
    }
  }, props)));
};

exports.FileInputButton = FileInputButton;
var _default = FileInputButton;
exports.default = _default;
