// sanity-plugin-mux-input/build/components/Preview.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _preview = require("part:@sanity/base/preview");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _secrets = require("../actions/secrets");

var _getPosterSrc = _interopRequireDefault(require("../util/getPosterSrc"));

var _Preview = _interopRequireDefault(require("./Preview.css"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr == null ? null : typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]; if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

var MuxVideoPreview = props => {
  var _React$useState = _react.default.useState(undefined),
      _React$useState2 = _slicedToArray(_React$useState, 2),
      secrets = _React$useState2[0],
      setSecrets = _React$useState2[1];

  var _React$useState3 = _react.default.useState(undefined),
      _React$useState4 = _slicedToArray(_React$useState3, 2),
      posterSrc = _React$useState4[0],
      setPosterSrc = _React$useState4[1];

  _react.default.useEffect(() => {
    (0, _secrets.fetchSecrets)().then(data => setSecrets(data.secrets));
  }, []);

  _react.default.useEffect(() => {
    if (props.value === null || secrets === undefined) return;
    if (props.value.status !== 'ready') return;
    var signingKeyId = secrets.signingKeyId,
        signingKeyPrivate = secrets.signingKeyPrivate;
    var playbackId = props.value.playbackIds[0];
    setPosterSrc((0, _getPosterSrc.default)(playbackId.id, {
      time: props.value.thumbTime,
      fitMode: 'crop',
      width: 640,
      height: 360,
      isSigned: playbackId.policy === 'signed',
      signingKeyId,
      signingKeyPrivate
    }));
  }, [props.value, secrets]);

  if (!props.value) return null;

  if (posterSrc !== undefined) {
    return /*#__PURE__*/_react.default.createElement("div", {
      className: _Preview.default.poster,
      style: {
        backgroundImage: "url(".concat(posterSrc, ")")
      }
    });
  }

  var _props$value = props.value,
      filename = _props$value.filename,
      playbackId = _props$value.playbackId,
      status = _props$value.status;
  return /*#__PURE__*/_react.default.createElement(_preview.SanityDefaultPreview, {
    title: filename || playbackId || '',
    subtitle: status ? "status: ".concat(status) : null
  });
};

MuxVideoPreview.propTypes = {
  value: _propTypes.default.object
};
MuxVideoPreview.defaultProps = {
  value: null
};
var _default = MuxVideoPreview;
exports.default = _default;
